zebra-consensus/src/chain/tests.rs

```diff
@@ -243,6 +243,10 @@ async fn verify_checkpoint() -> Result<(), Report> {
     /// SPANDOC: Make sure the verifier service is ready
     let ready_verifier_service = chain_verifier.ready_and().await.map_err(|e| eyre!(e))?;
 
+    // Add a timeout layer
+    let verifier_timeout = TimeoutLayer::new(Duration::from_secs(VERIFY_TIMEOUT_SECONDS));
+    verifier_timeout.layer(&ready_verifier_service);
+
     let transcript = Transcript::from(BLOCK_VERIFY_TRANSCRIPT_GENESIS.iter().cloned());
     transcript.check(ready_verifier_service).await.unwrap();
```

Comment:
Issue: Layer returns the wrapped service, so we need to run the transcript on the return value:
https://docs.rs/tower/0.3.1/tower/layer/trait.Layer.html#tymethod.layer

```suggestion
    // Add a timeout layer
    let ready_verifier_service = TimeoutLayer::new(Duration::from_secs(VERIFY_TIMEOUT_SECONDS)).layer(&ready_verifier_service);

    let transcript = Transcript::from(BLOCK_VERIFY_TRANSCRIPT_GENESIS.iter().cloned());
    transcript.check(ready_verifier_service).await.unwrap();
```

Reply:
Applied this at https://github.com/ZcashFoundation/zebra/pull/858/commits/73ebc537a99d4b51201bcc600263c9e8b6cc905b